enterprise_gateway/services/kernels/remotemanager.py

```diff
@@ -323,7 +323,11 @@ def remove_kernel(self, kernel_id: str) -> None:
         """
         Removes the kernel associated with `kernel_id` from the internal map and deletes the kernel session.
         """
-        super().remove_kernel(kernel_id)
+        try:
+            super().remove_kernel(kernel_id)
+        except web.HTTPError:  # this is hint for multiple shutdown request.
+            self.log.debug(f"Exception while removing kernel {kernel_id}: kernel not found.")
```

Comment:
Hmm - how can `super().remove_kernel(kernel_id)` raise anything other than `KeyError`?

What's odd is that the culling only operates off the entries in `self._kernels` so should never run into a missing key unless there's a timing issue where the key (kernel_id) is removed from `self._kernels` while a culling interval is taking place.

Is this easily reproduced in your environment?  It seems like it must require an issue to occur with the kernel during startup, since the culling is reporting a state of 'starting' - which means the kernel never completed its startup.

Reply:
Not easily reproducible

Reply:
yeah, I bet.  The only way to explain the stack trace is if the `kernel_id `key is removed from the `_kernels[]` list _after_ `shutdown_kernel()` has determined the `kernel_id` _does exist_.  As a result, I think the best course of action is to make `MultiKernelManager.remove_kernel()` tolerant of `KeyError` in `jupyter_client`.

Reply:
the issue with making this in the client is that there are backport pains. 

as for the key error, I see what you are saying, let me update that.